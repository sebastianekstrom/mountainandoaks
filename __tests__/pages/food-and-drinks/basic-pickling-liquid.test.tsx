import React from "react";
import { render, screen } from "@testing-library/react";
import BasicPicklingLiquid from "pages/food-and-drinks/basic-pickling-liquid";

describe("BasicPicklingLiquid page", () => {
  it("renders title correctly", () => {
    render(<BasicPicklingLiquid />);
    const title = screen.queryAllByText(/Basic pickling liquid/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<BasicPicklingLiquid />);
    const ingress = screen.queryAllByText(
      /A quick, flexible pickling base thatâ€™s easy to make and can be scaled for any amount of vegetables../i,
    );
    expect(ingress).toBeDefined();
  });

  it("renders the correct ingredients", () => {
    render(<BasicPicklingLiquid />);
    const ingredientList = screen.getByTestId("INGREDIENT_LIST");
    expect(ingredientList).toMatchSnapshot();
  });
});
