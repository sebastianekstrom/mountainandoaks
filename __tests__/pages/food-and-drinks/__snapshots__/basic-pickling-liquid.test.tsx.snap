// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicPicklingLiquid page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Champagne vinegar (or white wine vinegar)
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          200
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Sugar (granulated)
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          100
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Water
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          100
          g
        </p>
      </li>
    </ul>
  </div>
</section>
`;
